gobblin-data-management/src/main/java/gobblin/data/management/copy/replication/ConfigBasedDatasetsFinder.java

```diff
@@ -92,6 +114,8 @@ public ConfigBasedDatasetsFinder(Properties props) throws IOException {
         new Path(props.getProperty(GOBBLIN_REPLICATION_CONFIG_STORE_DATASET_COMMON_ROOT)));
     this.whitelistTag = PathUtils.mergePaths(new Path(this.storeRoot),
         new Path(props.getProperty(GOBBLIN_REPLICATION_CONFIG_STORE_WHITELIST_TAG)));
+    this.threadPoolSize = props.containsKey(THREAD_POOL_SIZE) ? Integer.parseInt(props.getProperty(THREAD_POOL_SIZE))
```

Comment:
I'd suggest instead extending `IterableDatasetFinder`. If you do that, `CopySource` will handle parallelization for you.